Replace deprecated `ember-cli/ext/promise` with `rsvp`

leizhao4 opened the pull request:
## What Changed & Why
- Replaced deprecated `ember-cli/ext/promise` with `rsvp`. Ember CLI 2.12 Beta complains about this deprecation.
- Bumped Node version in CI configs to 4 because 0.12 was not happy with `const`s.

brunowego commented:
@leizhao4 thanks for this. Very annoying these warnings on console.

lukemelia commented:
@leizhao4 Thanks! Would you mind squashing the 1st and 3rd commits together?

lukemelia commented:
Also, I think we should hold off on breaking node 0.12 compatibility, because this addon should still work with older versions of ember-cli, which support 0.12.

leizhao4 commented:
@lukemelia Sure! I just removed the 2nd commit and squashed the 1st and 3rd. But one dependency `username` no longer supports Node 0.12 so one test case will fail.

leizhao4 commented:
I reported Node 0.12 incompatibility to `broccoli-babel-transpiler`:

https://github.com/babel/broccoli-babel-transpiler/issues/111

lukemelia commented:
@leizhao4 Thanks for your work on this. I incorporated it into a larger change in #36 and will be releasing it as a major version bump.
